src/providers/data_provider_fo.c

```diff
@@ -24,6 +24,8 @@
 #include "providers/backend.h"
 #include "resolv/async_resolv.h"
 
+#define MAX_CURRENT_RESOLVED_ATTEMPTS 1
```

Comment:
I think either the name or a value is a little bit misleading as you actually want to try the server twice.